src/app/component/SliderWidget/SliderWidget.component.js

```diff
@@ -65,37 +64,22 @@ export class SliderWidget extends PureComponent {
         clearInterval(this.carouselInterval);
     }
 
-    onActiveImageChange = (activeImage) => {
-        this.setState({ activeImage });
+    onActiveSlideChange = (activeSlide) => {
+        this.setState({ activeSlide });
     };
 
     startCarousel = (interval) => {
         this.carouselInterval = setInterval(() => {
             this.getImageToShow();
-
-            const { imageToShow } = this.state;
-
-            this.onActiveImageChange(imageToShow);
         }, interval);
     };
 
     getImageToShow() {
-        const { activeImage, carouselDirection } = this.state;
-        const { slider: { slides } } = this.props;
-
-        if (activeImage === 0) {
-            this.setState({
-                carouselDirection: 'right',
-                imageToShow: activeImage + 1
-            });
-        } else if (activeImage === slides.length - 1) {
-            this.setState({
-                carouselDirection: 'left',
-                imageToShow: activeImage - 1
-            });
-        } else {
-            this.setState({ imageToShow: carouselDirection === 'right' ? activeImage + 1 : activeImage - 1 });
-        }
+        const { activeSlide } = this.state;
+        const { slider: { slides: { length } } } = this.props;
+
+        const toShow = activeSlide >= length - 1 ? 0 : activeSlide + 1;
+        this.setState({ activeSlide: toShow });
```

Comment:
This is not `getImageToShow`, rarther `updateImageToShow`.
Prefer using `getX` to return some values, not to mutate the state.